js/l1_player.js

```diff
@@ -0,0 +1,279 @@
+/* global localStorage */
+// eslint-disable-next-line no-unused-vars
+{
+  const proto = Object.getPrototypeOf(localStorage);
+  proto.getObject = function getObject(key) {
+    const value = this.getItem(key);
+    return value && JSON.parse(value);
+  };
+  proto.setObject = function setObject(key, value) {
+    this.setItem(key, JSON.stringify(value));
+  };
+  Object.setPrototypeOf(localStorage, proto);
+
+  const backgroundCall = (callback) => {
+    (chrome || browser).runtime.getBackgroundPage(callback);
+  };
+
+  const initPlayer = (player) => {
+    // add songs to playlist
+    let localCurrentPlaying = localStorage.getObject('playing-list');
+    if (localCurrentPlaying === null) {
+      localCurrentPlaying = localStorage.getObject('current-playing');
+    }
+    if (localCurrentPlaying === null) {
+      return;
+    }
+    player.setNewPlaylist(localCurrentPlaying);
+
+    const localPlayerSettings = localStorage.getObject('player-settings');
+    if (localPlayerSettings === null) {
+      return;
+    }
+    const track_id = localPlayerSettings.nowplaying_track_id;
+    player.loadById(track_id);
+  };
+
+  const l1Player = {
+    status: {
+      muted: false,
+      volume: 1,
+      loop_mode: 0,
+      playing: null,
+    },
+    bootstrapTrack: null,
+    play() {
+      backgroundCall((w) => {
+        w.player.play();
+      });
+    },
+    pause() {
+      backgroundCall((w) => {
+        w.player.pause();
+      });
+    },
+    toggleplayPause() {
+      backgroundCall((w) => {
+        if (w.player.playing) {
+          w.player.pause();
+        } else {
+          w.player.play();
+        }
+      });
+    },
+    playById(id) {
+      backgroundCall((w) => {
+        w.player.playById(id);
+      });
+    },
+    loadById(idx) {
+      backgroundCall((w) => {
+        w.player.loadById(idx);
+      });
+    },
+    seek(per) {
+      backgroundCall((w) => {
+        w.player.seek(per);
+      });
+    },
+    next() {
+      backgroundCall((w) => {
+        w.player.skip('next');
+      });
+    },
+    prev() {
+      backgroundCall((w) => {
+        w.player.skip('prev');
+      });
+    },
+    random() {
+      backgroundCall((w) => {
+        w.player.skip('random');
+      });
+    },
+    setLoopMode(input) {
+      backgroundCall((w) => {
+        w.player.setLoopMode(input);
+      });
+    },
+    mute() {
+      backgroundCall((w) => {
```

Comment:
It seems that l1player and player in player_thread share a lot of common methods, so maybe we can sync code and provide same method name for both object. It will be better to understand.

Reply:
Using the different method names is for the purpose of aware the developer they belong to different objects. This will reduce the misunderstood and incorrect invokes.

Reply:
Got it. I thought l1player is a "proxy" of player_thread, and if they're different object, it make sense